mlos_bench/mlos_bench/optimizer/mlos_core_optimizer.py

```diff
@@ -62,15 +72,19 @@ def suggest(self) -> TunableGroups:
     def register(self, tunables: TunableGroups, status: Status,
                  score: Union[float, dict] = None) -> float:
         score = super().register(tunables, status, score)
-        # By default, hyperparameters in ConfigurationSpace are sorted by name:
-        df_config = pd.DataFrame(dict(sorted(tunables.get_param_values().items())), index=[0])
-        _LOG.debug("Dataframe:\n%s", df_config)
-        self._opt.register(df_config, pd.Series([score]))
+        # TODO: mlos_core currently does not support registration of failed trials:
```

Comment:
As mentioned above - `mlos_core` shouldn't need to know about that.  It just handles scores and data.

`mlos_bench` however can compute some fabricated scores for failed trials for `mlos_core` based on the *real* data it's seen so far.

Reply:
That can be a future PR though.